lib/std/os/windows/kernel32.zig

```diff
@@ -411,3 +415,11 @@ pub extern "kernel32" fn SleepConditionVariableSRW(
 pub extern "kernel32" fn TryAcquireSRWLockExclusive(s: *SRWLOCK) callconv(WINAPI) BOOLEAN;
 pub extern "kernel32" fn AcquireSRWLockExclusive(s: *SRWLOCK) callconv(WINAPI) void;
 pub extern "kernel32" fn ReleaseSRWLockExclusive(s: *SRWLOCK) callconv(WINAPI) void;
+
+pub extern "kernel32" fn RegOpenKeyExW(
+    hkey: HKEY,
+    lpSubKey: LPCWSTR,
+    ulOptions: DWORD,
+    samDesired: REGSAM,
+    phkResult: *HANDLE,
```

Comment:
If this definition is going to stick around, [`phkResult` should be `*HKEY`](https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regopenkeyexw), but it looks like `RegOpenKeyExW` is not used by the current implementation.

Reply:
Good catch, I'll address it in this PR too. Funnily enough, I've been using this function as-is on my local Win11 installation and never run into any problems with mixing of HKEY and HANDLE types.

Reply:
That makes sense, since `HANDLE` is `*anyopaque`, it will gladly accept anything:

```zig
const std = @import("std");

pub fn main() void {
    const hkey = @intToPtr(std.os.windows.HKEY, 0x80000002);
    const handle: std.os.windows.HANDLE = hkey;

    std.debug.print("as HKEY: {*} as HANDLE: {*}\n", .{ hkey, handle });
}
```
outputs:
```
as HKEY: os.windows.HKEY__opaque_2600@80000002 as HANDLE: anyopaque@80000002
```

Reply:
Mhm, makes sense. I have now fixed the signature in 2528189